File: java/src/eti/Project2/Organisms/Belladonna.java
package eti.Project2.Organisms;

import eti.Project2.GUI;
import eti.Project2.MapPosition;
import eti.Project2.World;

/**
 * Created By IDEA
 * Author: s188026 Yifei Liu
 * Date: 2021/5/30 12:59
 * Description: Belladonna  '$'
 *      Kills any animal which eats it.
 */

public class Belladonna extends Plant{
    //constructor
    public Belladonna(char n, int s, int i, int X, int Y, World w) {
        super(n, s, i, X, Y, w);
    }

    public Belladonna(Belladonna b) {
        super(b);
    }

    public Belladonna(int X, int Y, World world){
        setName('$');
        setStrength(99);
        setInitiative(0);
        setPosition(new MapPosition(X, Y));
        setWorld(world);
    }

    @Override
    public void Collision(Organism attacker)
    {
        //who eats this will die(so this will be eaten)
        /**std::cout + "'" + getName() + "' kills attacker who eats it'" + (Grid(attacker.getPosition().getX(), attacker.getPosition().getY())).getName() + "'(" + attacker.getPosition().getX() + "," + attacker.getPosition().getY() + ")" + std::endl;*/
        GUI.getInformation().append("'" + getName() + "' kills attacker who eats it'" + attacker.getName() + "'(" + attacker.getPosition().getX() + "," + attacker.getPosition().getY() + ")\n");
        
        getWorld().organisms[getPosition().getY()][getPosition().getX()] = new Organism('_', -1, -1, getPosition().getX(), getPosition().getY(), getWorld());
    }
}

File: cpp/Belladonna.cpp
#include "Belladonna.h"
#include <iostream>

Belladonna::Belladonna(int X, int Y, World* world) :Plant(BELLADONNA, 99, 0, X, Y, world) {};

void Belladonna::Collision(Organism*& attacker)
{
	//who eats this will die(so this will be eaten)
	//Grid(attacker->GetPosition().X, attacker->GetPosition().Y) = new nullGrid(attacker->GetPosition());
	
	std::cout << "'" << GetName() << "' kills attacker who eats it'" << (Grid(attacker->GetPosition().X, attacker->GetPosition().Y))->GetName() << "'(" << attacker->GetPosition().X << "," << attacker->GetPosition().Y << ")" << std::endl;
	
	delete(attacker);

	Grid(GetPosition().X, GetPosition().Y) = new nullGrid(GetPosition());
	delete(this);
}
